qc_utils.py

```diff
@@ -947,14 +989,23 @@ def find_continent(country_code: str) -> str:
     return concord[country_code]
 
 #************************************************************************
-def prepare_data_repeating_string(data, diff=0, plots=False, diagnostics=False):
+def prepare_data_repeating_streak(data: np.array, diff:int = 0,
+                                  plots:bool = False, diagnostics:bool = False) -> tuple[np.array,
+                                                                                         np.array,
+                                                                                         np.array]:
     """
-    Prepare the data for repeating strings
+    Prepare the data for repeating streaks
 
     :param np.array data: data to assess
-    :param int diff: difference to look for (0 in strings of data, 1 in strings of indices)
+    :param int diff: difference to look for (0 in streaks of data, 1 in streaks of indices)
     :param bool plots: turn on plots
     :param bool diagnostics: turn on diagnostic output
+
+    :returns: tuple(array, array, array)
+
+    array of the streak lengths in temporal order
+    array of the grouped differences (difference, count)
```

Comment:
Not sure what grouped differences are?

Reply:
In this case it pairs of the difference between values (e.g. 0 for a repeating string) and the number of times this occurs before the difference changes.